activerecord/lib/active_record/database_configurations/connection_url_resolver.rb

```diff
@@ -67,14 +67,16 @@ def raw_config
               database: uri.opaque
             )
           else
-            query_hash.merge(
+            connection_attrs = {
               adapter: @adapter,
               username: uri.user,
               password: uri.password,
               port: uri.port,
               database: database_from_path,
               host: uri.hostname
-            )
+            }.compact_blank
+
+            query_hash.merge(connection_attrs)
```

Comment:
I think this would be nice case for using reverse_merge:
https://edgeapi.rubyonrails.org/classes/Hash.h. tml#method-i-reverse_merge
It is used in several place in the Rails code base. 
Than we could probably skip the compact_blank.

Reply:
```ruby
connection_attrs = {
  adapter: @adapter,
  username: uri.user, 
  password: uri.password,
  port: uri.port,
  database: database_from_path,
  host: uri.hostname
}.compact_blank

connection_attrs.reverse_merge(query_hash)
```

Did you mean something like this? Without `compact_blank` it isn't working, because `reverse_merge` doesn't reject `nil` values.

Reply:
Nice! I wasn’t sure it would work 😄

Reply:
Same. I first commented that it didn't work but later I realized that I just had to replace `merge` with `reverse_merge`. Silly of me 😁